src/test/java/org/broadinstitute/consent/http/authentication/OAuthAuthenticatorTest.java

```diff
@@ -99,13 +97,44 @@ void testAuthenticateGetUserWithStatusInfoFailurePostUserSuccess() throws Except
     verify(samService, times(1)).postRegistrationInfo(any());
   }
 
+  /**
+   * Test that in the case of a Sam user lookup failure, we then try to register the user.
+   * if that fails, we throw an exception.
+   */
+  @Test
+  void testAuthenticateGetUserWithStatusInfoFailurePostUserFailureWebAppEx() throws Exception {
+    headerMap.put(ClaimsCache.OAUTH2_CLAIM_email, List.of("email"));
+    headerCache.loadCache(bearerToken, headerMap);
+    when(samService.getRegistrationInfo(any())).thenThrow(new NotFoundException());
+    when(samService.postRegistrationInfo(any())).thenThrow(new WebApplicationException("errorMessage"));
+    oAuthAuthenticator = new OAuthAuthenticator(samService);
+
+    WebApplicationException ex = assertThrows(WebApplicationException.class, () -> oAuthAuthenticator.authenticate(bearerToken));
+    assertEquals("errorMessage", ex.getMessage());
+  }
+
+  /**
+   * Test that in the case of a Sam user lookup failure, we then try to register the user.
+   * if that fails, we throw an exception.
+   */
+  @Test
+  void testAuthenticateGetUserWithStatusInfoFailurePostUserFailureAuthEx() throws Exception {
+    headerMap.put(ClaimsCache.OAUTH2_CLAIM_email, List.of("email"));
+    headerMap.put(ClaimsCache.OAUTH2_CLAIM_name, List.of("name"));
+    headerCache.loadCache(bearerToken, headerMap);
+    when(samService.getRegistrationInfo(any())).thenThrow(new NotFoundException());
+    when(samService.postRegistrationInfo(any())).thenThrow(new AuthenticationException("errorMessage"));
+    oAuthAuthenticator = new OAuthAuthenticator(samService);
+
+    AuthenticationException ex = assertThrows(AuthenticationException.class, () -> oAuthAuthenticator.authenticate(bearerToken));
+    assertEquals("AuthUser not able to be registered: '{\"email\":\"email\",\"name\":\"name\"}", ex.getMessage());
```

Comment:
A text block can be used here
```suggestion
    assertEquals("""
        AuthUser not able to be registered: '{"email":"email","name":"name"}""", ex.getMessage());
```